add support for string module ids

jbellenger opened the pull request:
ProductionModePlugin writes module ids as unescaped values. This works for the most common case of integer module ids, but not for strings. Trying to use ProductionModePlugin with something like [NamedModulesPlugin](https://github.com/webpack/webpack/blob/master/lib/NamedModulesPlugin.js), renders the snippet below which throws a syntax error at runtime:

```javascript
module.exports = factory( __webpack_require__(./node_modules/globalize/dist/globalize-runtime/number.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/plural.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/message.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/currency.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/date.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/relative-time.js), __webpack_require__(./node_modules/globalize/dist/globalize-runtime/unit.js) );
```

This PR adds support for quoting module ids that look like strings.

rxaviers commented:
Any unit test we could add?

jbellenger commented:
Totally, though I think it might require bumping the webpack devDep to webpack2. Does that work?

rxaviers commented:
Sure, go for it... Any suggestions how we could test for various webpack versions? One approach would be splitting the plugin development into webpack-1 and webpack-2 separate branches...

jbellenger commented:
I'll update the PR to include a webpack2 upgrade.
As far as supporting different webpack versions, for simplicity sake I'd just bump the version to 0.4.0 and drop support for webpack1. I'm happy to include that in this PR if that works for you.

jbellenger commented:
Updated the diff to include tests for hashed and named module ids. While I was in there, I added linting for tests and moved the test output directory to allow running mocha with `--watch`

Example test output:
```
jbellenger !1008 14:36> npm run test                                                                                                                                                                                                                                                                                                                                                                
> globalize-webpack-plugin@0.3.13 test /Users/jbellenger/workspace/globalize-webpack-plugin
> eslint *js test && mocha test
  Globalize Webpack Plugin
    Production Mode
      when using default module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          ✓ should include formatDate
          ✓ should include formatNumber
          ✓ should include formatCurrency
          ✓ should include formatMessage
          ✓ should include formatRelativeTime
          ✓ should include formatUnit
          ✓ should include parseNumber
          ✓ should include parseDate
      when using named module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          ✓ should include formatDate
          ✓ should include formatNumber
          ✓ should include formatCurrency
          ✓ should include formatMessage
          ✓ should include formatRelativeTime
          ✓ should include formatUnit
          ✓ should include parseNumber
          ✓ should include parseDate
      when using hashed module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          ✓ should include formatDate
          ✓ should include formatNumber
          ✓ should include formatCurrency
          ✓ should include formatMessage
          ✓ should include formatRelativeTime
          ✓ should include formatUnit
          ✓ should include parseNumber
          ✓ should include parseDate


  27 passing (1s)
```

necolas commented:
> for simplicity sake I'd just bump the version to 0.4.0 and drop support for webpack1

SGTM too. The webpack globalize tools could commit to updating to webpack@2 (now at 2.5+) and drop webpack@1 altogether. It would simplify development and help move consumers of these packages onto the latest tools

rxaviers commented:
Thank you!!

rxaviers commented:
Released `1.0.0-alpha.0` pre-release (webpack-2 only support).
